src/main/java/de/fraunhofer/aisec/cpg/passes/CallResolver.java

```diff
@@ -673,29 +929,77 @@ private ConstructorDeclaration createConstructorDummy(
-        .map(FunctionDeclaration.class::cast)
-        .collect(Collectors.toList());
+        Pattern.compile(
+            "(" + Pattern.quote(recordDeclaration.getName()) + "\\.)?" + Pattern.quote(name));
+
+    if (lang instanceof CXXLanguageFrontend) {
+      List<FunctionDeclaration> invocationCandidate =
+          new ArrayList<>(
+              recordDeclaration.getMethods().stream()
+                  .filter(
+                      m -> namePattern.matcher(m.getName()).matches() && m.hasSignature(signature))
+                  .map(FunctionDeclaration.class::cast)
+                  .collect(Collectors.toList()));
+
+      if (invocationCandidate.isEmpty()) {
+        // Search for possible invocation with defaults args
+        invocationCandidate.addAll(
+            resolveWithDefaultArgs(
+                call,
+                recordDeclaration.getMethods().stream()
+                    .filter(
+                        m ->
+                            namePattern.matcher(m.getName()).matches()
+                                && !m.isImplicit()
+                                && call.getSignature().size() < m.getSignatureTypes().size())
+                    .map(FunctionDeclaration.class::cast)
+                    .collect(Collectors.toList())));
+      }
+
+      if (invocationCandidate.isEmpty()) {
+        // Search for possible invocation with implicit cast
+        invocationCandidate.addAll(
+            resolveWithImplicitCast(
+                call,
+                recordDeclaration.getMethods().stream()
+                    .filter(m -> namePattern.matcher(m.getName()).matches() && !m.isImplicit())
+                    .map(FunctionDeclaration.class::cast)
+                    .collect(Collectors.toList())));
+      }
+
+      return invocationCandidate;
+    } else {
+      return recordDeclaration.getMethods().stream()
+          .filter(m -> namePattern.matcher(m.getName()).matches() && m.hasSignature(signature))
+          .map(FunctionDeclaration.class::cast)
+          .collect(Collectors.toList());
+    }
   }
 
   private List<FunctionDeclaration> getInvocationCandidatesFromParents(
-      String name, List<Type> signature, Set<RecordDeclaration> possibleTypes) {
+      String name, CallExpression call, Set<RecordDeclaration> possibleTypes) {
+    Set<RecordDeclaration> workingPossibleTypes = new HashSet<>(possibleTypes);
     if (possibleTypes.isEmpty()) {
       return new ArrayList<>();
     } else {
       List<FunctionDeclaration> firstLevelCandidates =
           possibleTypes.stream()
-              .map(r -> getInvocationCandidatesFromRecord(r, name, signature))
+              .map(r -> getInvocationCandidatesFromRecord(r, name, call))
               .flatMap(Collection::stream)
               .collect(Collectors.toList());
-      if (firstLevelCandidates.isEmpty()) {
-        return possibleTypes.stream()
+
+      if (lang instanceof CXXLanguageFrontend) {
```

Comment:
Why is this specific for C++? Maybe a short comment would be good here

Reply:
Yes I will add a comment. As far as I know this only applies to C++. An example would be the following:
```
class Base
{
public:
    int calc(int i)
    {
        cout << "int: ";
        return i+6;
    }
};
class Overload : public Base
{
public:
    double calc(double d)
    {
        cout << "double: ";
        return d+9.9;
    }
};
int main()
{
    Overload overloaded;
    cout << overloaded.calc(1) << '\n';
    cout << overloaded.calc(1.1) << '\n';
    return 0;
}
```
In C++ function overloading cannot happen at different levels in the hierarchy. That is the reason why both of those calls will invoke the `double calc(double d)` function. In java the overload class would have both overloaded methods and therefore the first call would call the method with the `int` as parameter